File: http/http-server-annotation-processor/src/test/java/ru/tinkoff/kora/http/server/annotation/processor/controller/TestControllerPathParameters.java
package ru.tinkoff.kora.http.server.annotation.processor.controller;

import ru.tinkoff.kora.http.common.annotation.HttpRoute;
import ru.tinkoff.kora.http.common.annotation.Path;
import ru.tinkoff.kora.http.server.common.annotation.HttpController;

import static ru.tinkoff.kora.http.common.HttpMethod.GET;

@HttpController
public class TestControllerPathParameters {

    @HttpRoute(method = GET, path = "/pathString/{string}")
    public String pathString(@Path String string) {
        return string;
    }

    @HttpRoute(method = GET, path = "/pathStringWithName/{stringWithName}")
    void pathStringWithName(@Path("stringWithName") String string) {
    }

    @HttpRoute(method = GET, path = "/pathInteger/{value}")
    void pathInteger(@Path int value) {
    }

    @HttpRoute(method = GET, path = "/pathIntegerObject/{value}")
    void pathIntegerObject(@Path Integer value) {
    }

    @HttpRoute(method = GET, path = "/pathLong/{value}")
    void pathLong(@Path long value) {
    }

    @HttpRoute(method = GET, path = "/pathLongObject/{value}")
    void pathLongObject(@Path Long value) {
    }

    @HttpRoute(method = GET, path = "/pathDouble/{value}")
    void pathDouble(@Path double value) {
    }

    @HttpRoute(method = GET, path = "/pathDoubleObject/{value}")
    void pathDoubleObject(@Path Double value) {
    }

    @HttpRoute(method = GET, path = "/pathEnum/{value}")
    void pathEnum(@Path TestEnum value) {
    }
}

File: http/http-server-symbol-processor/src/test/kotlin/ru/tinkoff/kora/http/server/symbol/processor/controllers/TestControllerPathParameters.kt
package ru.tinkoff.kora.http.server.symbol.processor.controllers

import ru.tinkoff.kora.http.common.HttpMethod
import ru.tinkoff.kora.http.common.annotation.HttpRoute
import ru.tinkoff.kora.http.common.annotation.Path
import ru.tinkoff.kora.http.server.common.annotation.HttpController

@HttpController
open class TestControllerPathParameters {
    @HttpRoute(method = HttpMethod.GET, path = "/pathString/{string}")
    open fun pathString(@Path string: String): String {
        return string
    }

    @HttpRoute(method = HttpMethod.GET, path = "/pathStringWithName/{stringWithName}")
    open fun pathStringWithName(@Path("stringWithName") string: String?) {
    }

    @HttpRoute(method = HttpMethod.GET, path = "/pathInteger/{value}")
    open fun pathInteger(@Path value: Int) {
    }

    @HttpRoute(method = HttpMethod.GET, path = "/pathIntegerObject/{value}")
    open fun pathIntegerObject(@Path value: Int?) {
    }

    @HttpRoute(method = HttpMethod.GET, path = "/pathLong/{value}")
    open fun pathLong(@Path value: Long) {
    }

    @HttpRoute(method = HttpMethod.GET, path = "/pathLongObject/{value}")
    open fun pathLongObject(@Path value: Long?) {
    }

    @HttpRoute(method = HttpMethod.GET, path = "/pathDouble/{value}")
    open fun pathDouble(@Path value: Double) {
    }

    @HttpRoute(method = HttpMethod.GET, path = "/pathDoubleObject/{value}")
    open fun pathDoubleObject(@Path value: Double?) {
    }

    @HttpRoute(method = HttpMethod.GET, path = "/pathEnum/{value}")
    open fun pathEnum(@Path value: TestEnum?) {
    }
}
